src/address.rs

```diff
@@ -1,116 +1,81 @@
-        let sp = [sn_size, addr_size, sn.as_slice(), addr.as_slice()].concat();
-        // include in fixed-length container
-        let mut key = [0u8; MAX_ADDRESS_LEN];
-        key[..sp.len()].copy_from_slice(sp.as_slice());
         Ok(Self {
-            inner: Address::new_delegated(*DEFAULT_IPC_ACTOR_NAMESPACE_ID, &key)?,
+            subnet_id: sn.clone(),
+            raw_address: *addr,
         })
     }
 
     /// Generates new IPC address
     pub fn new_from_ipc(sn: &SubnetID, addr: &Self) -> Result<Self, Error> {
-        let sn = sn.to_bytes();
-        let addr = addr.to_bytes();
-        let sn_size_vec = to_leb_bytes(sn.len() as u64)?;
-        let sn_size: &[u8] = sn_size_vec.as_ref();
-        let addr_size_vec = to_leb_bytes(addr.len() as u64)?;
-        let addr_size: &[u8] = addr_size_vec.as_ref();
-        let sp = [sn_size, addr_size, sn.as_slice(), addr.as_slice()].concat();
-        // include in fixed-length container
-        let mut key = [0u8; MAX_ADDRESS_LEN];
-        key[..sp.len()].copy_from_slice(sp.as_slice());
         Ok(Self {
-            inner: Address::new_delegated(*DEFAULT_IPC_ACTOR_NAMESPACE_ID, &key)?,
+            subnet_id: sn.clone(),
+            raw_address: addr.raw_address,
         })
     }
 
     /// Returns subnets of a IPC address
     pub fn subnet(&self) -> Result<SubnetID, Error> {
-        let bz = self.inner.payload().to_raw_bytes();
-        let sn_size = from_leb_bytes(&[bz[0]]).unwrap() as usize;
-        let s =
-            String::from_utf8(bz[2..sn_size + 2].to_vec()).map_err(|_| Error::InvalidIPCAddr)?;
-        SubnetID::from_str(&s).map_err(|_| Error::InvalidIPCAddr)
+        Ok(self.subnet_id.clone())
     }
 
     /// Returns the raw address of a IPC address (without subnet context)
     pub fn raw_addr(&self) -> Result<Address, Error> {
-        let bz = self.inner.payload().to_raw_bytes();
-        let sn_size = from_leb_bytes(&[bz[0]]).unwrap() as usize;
-        Ok(Address::from_bytes(&bz[2 + sn_size..])?)
+        Ok(self.raw_address)
     }
 
     /// Returns encoded bytes of Address
-    pub fn to_bytes(&self) -> Vec<u8> {
-        self.inner.to_bytes()
+    pub fn to_bytes(&self) -> Result<Vec<u8>, Error> {
+        Ok(cbor::serialize(self, "ipc-address")?.to_vec())
     }
-}
 
-pub(crate) fn to_leb_bytes(id: u64) -> Result<Vec<u8>, Error> {
-    // write id to buffer in leb128 format
-    Ok(unsigned_varint::encode::u64(id, &mut unsigned_varint::encode::u64_buffer()).into())
-}
+    pub fn from_bytes(bz: &[u8]) -> Result<Self, Error> {
+        let i: Self = cbor::deserialize(&RawBytes::new(bz.to_vec()), "ipc-address")?;
+        Ok(i)
+    }
 
-pub(crate) fn from_leb_bytes(bz: &[u8]) -> Result<u64, Error> {
-    // write id to buffer in leb128 format
-    let (id, remaining) = unsigned_varint::decode::u64(bz)?;
-    if !remaining.is_empty() {
-        return Err(Error::InvalidPayload);
+    pub fn to_string(&self) -> Result<String, Error> {
```

Comment:
We should make the serialization "human readable". I would suggest building the string by appending the SubnetID to the string representation of the address: `<raw_address>/<string address>`.

Reply:
Good point with the delimited, that was my bad. In previous versions we used `:` like `/root/t01:f02..`. If you don't mind change it to `:` instead of `-`? Thanks!

Reply:
Also, I would suggest adding the delimiter in a const so we don't have to go through the code figuring out where we used it if we change it in the future.